packages/framework/src/parser.ts

```diff
@@ -67,18 +69,22 @@ export const EnsembleParser = {
   ): EnsembleScreenModel | EnsembleMenuModel => {
     const view = get(screen, "View");
     const viewNode = get(view, "body");
-    const header = get(view, "header") as Record<string, unknown> | undefined;
+    const header = get(view, "header") as EnsembleHeaderModel | undefined;
+    const footer = get(view, "footer") as EnsembleFooterModel | undefined;
```

Comment:
The parser is meant to take raw YAML and assert that it matches the typed objects, not just cast them directly.